523: UI updates for the submission steps on "Skill" & "Knowledge" Review forms

adigidh opened the pull request:
fixes https://github.com/instructlab/ui/issues/523

Summary of changes on both skill and knowledge review sections:
- Update typography across the review submission step (font styles, and sizes) o
- Added content changes as requested in the figma
- Updated seed data nested component to display accordion style UX
- Updated component with semantic HTML

Screen recording ZIP:
[instructlab.mov.zip](https://github.com/user-attachments/files/18773796/instructlab.mov.zip)

Misjohns reviewed (changes requested):
Left my comments in previous message.

adigidh commented:
Thank you for taking a look. Made the following updates @Misjohns :

1. Improved line height for seed data accordion content 
2. The accordion header should match content frame color unless the user is hovering over the header. This was actually directly coming from the patteenfly component base styles. Overriden these styles now.

 <img width="1156" alt="image" src="https://github.com/user-attachments/assets/8d0bfaed-90c0-4324-a265-25e65925968a" />

3. Added `cursor: pointer` for hover on accordion header and content.

https://github.com/user-attachments/assets/03bc6027-e512-4ee0-97e4-e222e214ff4f

vishnoianil commented:
@adigidh Thanks for the PR, I pulled it locally for testing and overall it looks good.

Skill review form looks great!. 

In the knowledge review form, When i load Review Submission, browser throws the following error. Looks like you need to use unique key for each list element.
```
Each child in a list should have a unique "key" prop.
Check the render method of `ReviewSubmission`. See https://react.dev/link/warning-keys for more information.
```

vishnoianil commented on src/components/Contribute/Knowledge/ReviewSubmission/index.tsx:
```diff
@@ -2,114 +2,131 @@
-      </p>
+      <article>
+        <div className="info-wrapper">
+          <p className="submission-titles">Author Information</p>
```
To keep the same format across skill and knowledge review form, can you please update Author Information section to use "Contributor Information/Contributors" and add the relevant context text.

vishnoianil commented on src/components/Contribute/Knowledge/ReviewSubmission/index.tsx:
```diff
@@ -2,114 +2,131 @@
-      </p>
+      <article>
+        {/* File Path Information */}
+        <h5 className="category-titles">File Path Information</h5>
```
Let's use Directory Path, similar to skill form.

vishnoianil commented on src/components/Contribute/Skill/ReviewSubmission/index.tsx:
```diff
@@ -2,85 +2,107 @@
+    <>
+      <section className="review-submission-container">
+        <Content component={ContentVariants.h3}>Review</Content>
+        <p>Review the information below and click finish to submit your knowledge contribution. Use the back button to make changes.</p>
```
This is skill review form, so we need to remove all the reference for "Knowledge" through out this form.

vishnoianil commented on src/components/Contribute/Skill/ReviewSubmission/index.tsx:
```diff
@@ -2,85 +2,107 @@
+        {/* Seed Examples */}
+        <article>
+          <div className="info-wrapper">
+            <p>Seed data</p>
```
Let's use "Seed Example" here.

adigidh commented:
Thank you for taking a look, and the detailed review.
Summary of changes: @vishnoianil 

* Resolved duplicated key warning issues. Added unique keys for accordion items and used React.Fragment for unique key for QA pairs.
* Updated Author Information section to use "Contributor Information/Contributors"
* Changed File path to “Directory Path”
* Added fallback check for seed examples data
* Correct typo per this comment: https://github.com/instructlab/ui/pull/581/files/ba00d62d3a3a3e68c96bd093675386bebae86217#diff-3a687fbb5e4c1f2cd3417d6c54f9137c3a28704dcb7304898d851bdee57b74bf
* Updated OC -> DCO in content
* Removed all references for "Knowledge" through out the skill form
* "Seed Example"  used as label text

Can we create a separate tech debt issue to combine these components, add a unit/ integration test plus some documentation ? That will be beneficial to manage this in the future.

jeff-phillips-18 commented on src/components/Contribute/Skill/ReviewSubmission/index.tsx:
```diff
@@ -2,85 +2,107 @@
 import { Content, ContentVariants } from '@patternfly/react-core';
 import React from 'react';
+import './submission.css';
+import { Accordion, AccordionContent, AccordionItem, AccordionToggle } from '@patternfly/react-core';
```
Add these to the imports from `@patternfly/react-core`  above

jeff-phillips-18 commented on src/components/Contribute/Skill/ReviewSubmission/index.tsx:
```diff
@@ -2,85 +2,107 @@
+        {/* Author Information */}
+        <article>
+          <div className="info-wrapper">
+            <p className="submission-titles">Contributor Information</p>
```
We should use the `Content` components here as well

jeff-phillips-18 commented on src/components/Contribute/Skill/ReviewSubmission/submission.css:
```diff
@@ -0,0 +1,49 @@
+.review-submission-container {
+  .contributors-wrapper {
+    line-height: 2;
+    padding-top: 15px;
```
```suggestion
    padding-top: var(--pf-t--global--spacer--md);
```

jeff-phillips-18 commented on src/components/Contribute/Skill/ReviewSubmission/submission.css:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  .info-wrapper {
+    padding-bottom: 15px;
```
```suggestion
    padding-bottom: var(--pf-t--global--spacer--md);
```

jeff-phillips-18 commented on src/components/Contribute/Skill/ReviewSubmission/submission.css:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  .submission-titles {
+    font-size: var(--pf-t--global--font--size--sm) !important;
```
Can we use specificity rather than `!important`?

jeff-phillips-18 commented on src/components/Contribute/Skill/ReviewSubmission/submission.css:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  .accordion-wrapper {
+    padding-bottom: 15px;
```
```suggestion
    padding-bottom: var(--pf-t--global--spacer--md);
```

jeff-phillips-18 commented on src/components/Contribute/Skill/ReviewSubmission/submission.css:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  .category-titles {
+    font-weight: 600;
```
```suggestion
    font-weight: var(pf-t--global--font--weight--heading);
```

jeff-phillips-18 commented on src/components/Contribute/Skill/ReviewSubmission/submission.css:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  .accordion-content {
+    padding: 15px;
```
```suggestion
    padding: var(--pf-t--global--spacer--md);
```

jeff-phillips-18 commented on src/components/Contribute/Skill/ReviewSubmission/submission.css:
```diff
@@ -0,0 +1,49 @@
+  .seed-category-titles {
+    font-weight: 600;
+    font-size: var(--pf-t--global--font--size--xs) !important;
+    padding-bottom: 5px;
```
```suggestion
    padding-bottom: var(--pf-t--global--spacer--sm);
```

jeff-phillips-18 commented on src/components/Contribute/Skill/ReviewSubmission/submission.css:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  .seed-category-titles {
+    font-weight: 600;
```
```suggestion
    font-weight: var(pf-t--global--font--weight--heading);
```

adigidh commented:
Thank you taking a look @jeff-phillips-18 . Pushing the updates on the PR.

I appreciate the effort in ensuring consistency across components. Will this approach be standardized across all components moving forward? If so, it might be helpful to document this in our contribution guidelines or implement pre-commit or linting checks to enforce these patterns. Example: we can either enforce patternfly usage with ESLINT or create custom ESLint rules, or use pre commit hooks with husky. 

That said, I want to flag that the current updates were primarily focused on styling adjustments and adding in the accordion bits, without modifying the underlying elements. Since the existing components in the [main branch](https://github.com/instructlab/ui/blob/main/src/components/Contribute/Knowledge/ReviewSubmission/index.tsx#L16) don’t all follow this pattern of using PatternFly’s components, incorporating such changes could introduce scope creep in issues by adding fixes for technical debt alongside styling updates.

IMO it makes sense to address these tech debt improvements separately to keep the scope focused.

vishnoianil commented:
@adigidh can you please rebase your PR on main now ?

adigidh commented:
> @adigidh can you please rebase your PR on main now ?

@vishnoianil : just rebased and verified the new changes from main are in place and in sync with this PR.
